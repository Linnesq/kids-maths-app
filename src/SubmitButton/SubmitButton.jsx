import React from 'react';
import PropTypes from 'prop-types';
import './SubmitButton.scss';

const SubmitButton = (props) => {
    return (
        <div className="SubmitButton">
            <button data-testid="submitButton" type="submit" onClick={props.onClickHandler}>Submit</button>
        </div>
    )
};

SubmitButton.propTypes = {
    onClickHandler: PropTypes.func,
};

export default SubmitButton;