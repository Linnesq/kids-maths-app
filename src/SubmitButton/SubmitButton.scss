@import "../variables.scss";

.SubmitButton {
  margin-top: 10px;

  button {
    width: 200px;
    height: 40px;
    font-size: $textRegular;
    background-color: lightgreen;
  }
}
