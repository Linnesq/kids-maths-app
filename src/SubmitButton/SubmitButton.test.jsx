import React from 'react';
import { render, fireEvent } from '@testing-library/react';

import SubmitButton from './SubmitButton';

test('clicking on submit button should trigger handler function', () => {
    const mockHandler = jest.fn();
    const { container, getByTestId } = render(<SubmitButton onClickHandler={mockHandler}/>);
    expect(mockHandler).not.toHaveBeenCalled();
    
    fireEvent.click(getByTestId('submitButton'));
    expect(mockHandler).toHaveBeenCalledTimes(1);
    expect(container).toMatchSnapshot();
});