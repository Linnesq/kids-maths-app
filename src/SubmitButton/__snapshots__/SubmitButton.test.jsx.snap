// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicking on submit button should trigger handler function 1`] = `
<div>
  <div
    class="SubmitButton"
  >
    <button
      data-testid="submitButton"
      type="submit"
    >
      Submit
    </button>
  </div>
</div>
`;
